lib/libc/armstdc/include/sys/types.h

```diff
@@ -15,6 +15,8 @@
 #ifndef ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_SYS_TYPES_H_
 #define ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_SYS_TYPES_H_
 
+typedef unsigned int mode_t;
```

Comment:
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html says this should be a signed (not unsigned) type.

Reply:
I took that from `lib/libc/minimal/include/sys/types.h`.  So happy to change it to be signed.

Reply:
sounds like minimal libc has a bug; the spec says 'signed'. I expected it to be 'int', given the history of the APIs it gets used in (creat, mkdir) -- they came from a time before typedef when everything was 'int' :-)

Reply:
Hmm, that spec just says its an integer type (and I assume that since later they call out signed v unsigned, the spec would say 'signed integer' if that was the case).  Looking at linux headers they treat it as unsigned as well.

Reply:
good point; I read that as 'integer type' vs 'unsigned integer type', but you're right -- there are explicit notes for 'signed integer type' too. Either way looks fine to me.